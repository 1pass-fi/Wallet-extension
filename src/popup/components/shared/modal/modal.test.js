import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import Modal from '.'

describe('Test for  Modal component', () => {
  describe('Render without crashing', () => {
    let container, props, Children

    beforeEach(() => {
      Children = () => <h1 className='children'>Hello</h1>
      props = {
        children: <Children />
      }
      const obj = render(<Modal children={props.children} />)

      container = obj.container
      
    })

    it('renders correctly', () => {
      expect(container).toMatchSnapshot()
    })

    it('renders children', () => {
      expect(container.getElementsByClassName('children')[0].textContent).toEqual('Hello')
    })
  })
})
