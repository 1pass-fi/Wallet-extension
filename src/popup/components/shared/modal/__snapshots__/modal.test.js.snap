// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for  Modal component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="modal-container"
  >
    <div
      class="modal"
    >
      <div
        class="modal-close-icon"
      >
        <svg />
      </div>
      <h1
        class="children"
      >
        Hello
      </h1>
    </div>
  </div>
</div>
`;
